utils/zerovec/src/zerovec/mod.rs

```diff
@@ -312,10 +313,14 @@ where
         let capacity = vec.capacity();
         let len = vec.len();
         let ptr = mem::ManuallyDrop::new(vec).as_mut_ptr();
+        // Note: starting in 1.70 we can use NonNull::slice_from_raw_parts
         let slice = ptr::slice_from_raw_parts_mut(ptr, len);
         Self {
             vector: EyepatchHackVector {
-                buf: slice,
+                // Safety: `ptr` comes from Vec::as_mut_ptr, which says:
+                // "Returns an unsafe mutable pointer to the vector’s buffer,
+                // or a dangling raw pointer valid for zero sized reads"
+                buf: unsafe { NonNull::new_unchecked(slice) },
```

Comment:
Is this safety comment valid? If the vector is empty, is it guaranteed that the dangling pointer is non-null?

Reply:
^ @Manishearth you approved the PR but can you confirm my question about the dangling pointer being non-null?

Reply:
Yes, it is.